RecoParticleFlow/PFRecHitProducer/test/pfRecHitProducerAlpakaValidation.py

```diff
@@ -0,0 +1,386 @@
+    numberOfConcurrentRuns = cms.untracked.uint32(1),
+    numberOfStreams = cms.untracked.uint32(0),
+    numberOfThreads = cms.untracked.uint32(1),
+    printDependencies = cms.untracked.bool(False),
+    sizeOfStackForThreadsInKB = cms.optional.untracked.uint32,
+    throwIfIllegalParameter = cms.untracked.bool(True),
+    wantSummary = cms.untracked.bool(False)
+)
+
+# Production Info
+process.configurationMetadata = cms.untracked.PSet(
+    annotation = cms.untracked.string('reHLT nevts:5'),
+    name = cms.untracked.string('Applications'),
+    version = cms.untracked.string('$Revision: 1.19 $')
+)
+
+# Output definition
+process.FEVTDEBUGHLToutput = cms.OutputModule("PoolOutputModule",
+    dataset = cms.untracked.PSet(
+        dataTier = cms.untracked.string('GEN-SIM-DIGI-RAW'),
+        filterName = cms.untracked.string('')
+    ),
+    fileName = cms.untracked.string('reHLT_HLT.root'),
+    outputCommands = process.FEVTDEBUGHLTEventContent.outputCommands,
+    splitLevel = cms.untracked.int32(0)
+)
+
+# Other statements
+from HLTrigger.Configuration.CustomConfigs import ProcessName
+process = ProcessName(process)
+
+from Configuration.AlCa.GlobalTag import GlobalTag
+process.GlobalTag = GlobalTag(process.GlobalTag, 'auto:phase1_2022_realistic', '')
+
+# Path and EndPath definitions
+process.endjob_step = cms.EndPath(process.endOfProcess)
+process.FEVTDEBUGHLToutput_step = cms.EndPath(process.FEVTDEBUGHLToutput)
+
+# Schedule definition
+# process.schedule imported from cff in HLTrigger.Configuration
+process.schedule.extend([process.endjob_step,process.FEVTDEBUGHLToutput_step])
+from PhysicsTools.PatAlgos.tools.helpers import associatePatAlgosToolsTask
+associatePatAlgosToolsTask(process)
+
+# customisation of the process
+from HLTrigger.Configuration.customizeHLTforMC import customizeHLTforMC
+process = customizeHLTforMC(process)
+
+# Add early deletion of temporary data products to reduce peak memory need
+from Configuration.StandardSequences.earlyDeleteSettings_cff import customiseEarlyDelete
+process = customiseEarlyDelete(process)
+
+process.load( "HLTrigger.Timer.FastTimerService_cfi" )
+if 'MessageLogger' in process.__dict__:
+    process.MessageLogger.TriggerSummaryProducerAOD = cms.untracked.PSet()
+    process.MessageLogger.L1GtTrigReport = cms.untracked.PSet()
+    process.MessageLogger.L1TGlobalSummary = cms.untracked.PSet()
+    process.MessageLogger.HLTrigReport = cms.untracked.PSet()
+    process.MessageLogger.FastReport = cms.untracked.PSet()
+    process.MessageLogger.ThroughputService = cms.untracked.PSet()
+    process.MessageLogger.cerr.FastReport = cms.untracked.PSet( limit = cms.untracked.int32( 10000000 ) )
+
+
+#####################################
+##   Read command-line arguments   ##
+#####################################
+import sys
+import argparse
+parser = argparse.ArgumentParser(prog=f"{sys.argv[0]} {sys.argv[1]} --", description='Test and validation of PFRecHitProducer with Alpaka')
```

Comment:
I do not have this PR available testing, but I changed it so that arguments to `cmsRun` are allowed. Arguments to the config still have to separated by `--`. I think this should also work when including the other PR.

Reply:
The https://github.com/cms-sw/cmssw/pull/42650 was merged last week and will thus be part of the 13_3_0_pre4 that should be built soon.

> Arguments to the config still have to separated by `--`.

Did you add this requirement in order to the configuration to work in some way with #42650 as well? I'm asking because with #42650 the `--` is not needed anymore to separate the `cmsRun` arguments from the configuration file arguments (basically the arguments before the configuration file are used for `cmsRun`, and the arguments after the configuration file are passed in `sys.argv`). I'd suggest to revise the argument handling of this configuration file e.g. after the pre4 has been built, but I'd be ok if that is done in a subsequent PR (since this configuration file is not used in any unit tests).

Reply:
Originally, I added this because otherwise arguments would always be interpreted as begin for `cmsRun`. Since `cmsRun` ignores anything after `--`, this was the easiest method I found to pass arguments to the config. With the new PR, the `--` is not needed anymore, I think it should still work if it is present. I agree that this should be changed at some point.

Reply:
I switched to `CMSSW_13_3_0_pre4`, where #42650 is available and implemented this suggestion.